src/strands/agent/conversation_manager/recover_tool_use_on_max_tokens_reached.py

```diff
@@ -0,0 +1,66 @@
+"""Shared utility for handling token limit recovery in conversation managers."""
+
+import logging
+from typing import TYPE_CHECKING
+
+from ...hooks import MessageAddedEvent
+from ...types.content import ContentBlock, Message
+from ...types.exceptions import MaxTokensReachedException
+from ...types.tools import ToolUse
+
+if TYPE_CHECKING:
+    from ...agent.agent import Agent
+
+logger = logging.getLogger(__name__)
+
+
+def recover_tool_use_on_max_tokens_reached(agent: "Agent", exception: MaxTokensReachedException) -> None:
+    """Handle MaxTokensReachedException by cleaning up orphaned tool uses and adding corrected message.
+
+    This function fixes incomplete tool uses that may occur when the model's response is truncated
+    due to token limits. It:
+
+    1. Inspects each content block in the incomplete message for invalid tool uses
+    2. Replaces incomplete tool use blocks with informative text messages
+    3. Preserves valid content blocks in the corrected message
+    4. Adds the corrected message to the agent's conversation history
+
+    Args:
+        agent: The agent whose conversation will be updated with the corrected message.
+        exception: The MaxTokensReachedException containing the incomplete message.
+    """
+    logger.info("handling MaxTokensReachedException - inspecting incomplete message for invalid tool uses")
+
+    incomplete_message: Message = exception.incomplete_message
```

Comment:
Should we pass in this message instead of taking in an exception -  e.g. what does the exception get us?